plugin/lspconfig.lua

```diff
@@ -110,36 +110,47 @@ end, {
 
 api.nvim_create_user_command('LspStop', function(info)
   local current_buf = vim.api.nvim_get_current_buf()
-  local server_id, force, server_name
+  local server_id, force, server_name, err_msg
   local arguments = vim.split(info.args, '%s')
+
+  local filter = function()
+    return true
+  end
+  local found = true
+
   for _, v in pairs(arguments) do
     if v == '++force' then
       force = true
     elseif v:find '^[0-9]+$' then
-      server_id = v
-    else
+      server_id = tonumber(v)
+      ---@param client vim.lsp.Client
+      filter = function(client)
+        return server_id == client.id
+      end
+      found = false
+      err_msg = ('nvim-lspconfig: server id "%s" not found'):format(server_id)
+    elseif v ~= '' then
       server_name = v
-    end
-  end
-
-  if not server_id then
-    local servers_on_buffer = require('lspconfig.util').get_lsp_clients { bufnr = current_buf }
-    local found = false
-    for _, client in ipairs(servers_on_buffer) do
-      if client.attached_buffers[current_buf] and (server_name and (server_name == client.config.name) or true) then
-        client.stop(force)
-        found = true
+      ---@param client vim.lsp.Client
+      filter = function(client)
+        return server_name == client.config.name
       end
+      err_msg = ('nvim-lspconfig: server name "%s" not found'):format(server_name)
```

Comment:
it's not a "server" name. it's a config name. the server name is unrelated...

```suggestion
      err_msg = ('nvim-lspconfig: config "%s" not found'):format(server_name)
```